src/core/math/vec3/unit.js

```diff
@@ -0,0 +1,14 @@
+module.exports = unit
+const fromValues = require('./fromValues')
+const length = require('./length')
+
+/**
+ * Calculates the unit length of a vec3
+ *
+ * @param {vec3} a vector to calculate unit length of
+ * @returns {Number} unit length of a
+ */
+function unit (a) {
```

Comment:
- I feel stupid, but why do we need to pass in a length to a unit vector ?

Reply:
another remark but that is 100% my fault, the very short variable names originating from gl-vecXX are too short and non comprehensible for my taste (I have started to replace some). so feel free to replace `a`  & `l` with something more clear :)

Reply:
Actually, i'm not 100% as to which 'length' is being observed, i.e. what is 'w'?. I think this is the "unit vector" which represents the normal and tagent to perpendicular, but I could be wrong.

http://mathworld.wolfram.com/NormalVector.html

Do you know?